best-practice-certificate-provisioning.md

```diff
@@ -0,0 +1,313 @@
+
+Implementation of [BCP-003-01](best-practice-secure-comms.md) is a recommended prerequisite to implementing this document.
+
+## Use of Normative Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119][RFC-2119].
+
+## Normative References
+
+These appear at the end of the Markdown source for this document,
+and are referenced as hyperlinks within the main body.
+
+## Definitions
+
+_See also the [NMOS Glossary](https://github.com/AMWA-TV/nmos/wiki/Glossary), and definitions within RFCs._
+
+### API
+
+An HTTP / WebSocket API as defined in an AMWA NMOS Specification (IS-04, IS-05, IS-06, etc.)
+
+### CSR (Certificate Signing Request)
+
+The syntax used to request a TLS Certificate from a Certificate Authority, it contains the public key and identification information required by the Certificate Authority in the PKCS Format, [RFC 2986][RFC-2986]
+
+### EST (Enrollment over Secure Transport)
+
+Enrollment over Secure Transport defines a profile for requesting TLS certificates over a secure transport protocol.
+
+### EST Server
+
+The entity that is providing the EST API in accordance with [RFC 7030][RFC-7030]
+
+### EST Client
+
+The entity that is using the EST API, for example:
+
+- a Node requesting a TLS Server Certificate
+- a Node requesting the latest Root CA of the domain
+
+### NMOS Server
+
+The entity that is providing an NMOS API, for example:
+
+- a registry implementing IS-04 Registration and Query APIs
+- a Node implementing IS-04 Node API and IS-05 Connection API
+
+### NMOS Client
+
+The entity that is using the NMOS API, for example:
+
+- a Node using the IS-04 Registration API
+- a monitoring application using the IS-04 Query API
+- a connection control application using the IS-05 Connection API
+
+## Introduction (informative)
+
+This document covers the automated provisioning of TLS Certificates to NMOS Servers, which are then used to secure communications between NMOS Servers and NMOS Clients.
+
+This document is not concerned with the security of the connection used to carry out provisioning of the TLS Certificate, but for the mechanism described in this document to be effective the connection must be secured, ideally using the recommendation covered in [BCP-003-01](best-practice-secure-comms.md).
+
+## Automated Certificate Provisioning Flow (informative)
+
+To enable zero-configuration TLS Certificate provisioning, manufacturers must include a unique TLS Client Certificate on all their devices and provide the corresponding Root Certificate Authority to customers. If the TLS Client Certificate or the chain of trust is compromised the manufacturer must revoke the comprised certificates.
+
+1. Before the NMOS Node(EST Client) is shipped from the factory it must be provisioned with a unique TLS Client Certificate, signed by the Manufacturers Certificate Authority
```

Comment:
Perhaps the language could be modified slightly to account for pure software implementations?

Reply:
Slightly re-worded, but kept the word `manufacturer` as couldn't think of a better word that would cover both, software and hardware.